js/src/threading/ExclusiveData.h

```diff
@@ -109,10 +109,9 @@ class ExclusiveData {
   explicit ExclusiveData(const MutexId& id, Args&&... args)
       : lock_(id), value_(std::forward<Args>(args)...) {}
 
-  ExclusiveData(ExclusiveData&& rhs)
-      : lock_(std::move(rhs.lock)), value_(std::move(rhs.value_)) {
-    MOZ_ASSERT(&rhs != this, "self-move disallowed!");
-  }
+// Removed move constructor that both had a misspelled variable and attempted to move a Mutex, which
+// contains a member variable that deletes its move constructor.
+// TODO(SERVER-90310): When we next upgrade mozjs, we should revisit this modification.
```

Comment:
Just delete and don't leave a comment, I think. The git history will show the story behind the local mod.
We know local mods need to be revisited on upgrade already without the TODO.

If you wanted to keep a perfect description of the code like this you could `#ifdef` it out.
Then we would have no conflicts when upstream updates come in.
I think it's easier to just remove this entirely though.

```suggestion
```